app.rb

```diff
@@ -0,0 +1,114 @@
+require_relative 'student'
+require_relative 'person'
+require_relative 'teacher'
+require_relative 'book'
+require_relative 'rental'
+class App
+  def initialize
+    @books = []
+    @people = []
+    @rentals = []
+  end
+
+  def list_all_books
+    @books.each do |book|
+      puts "Title: '#{book.title}', Author: '#{book.author}'"
+    end
+  end
+
+  def list_all_peaple
+    @people.each do |person|
+      puts "[#{person.class}] Name:#{person.name}, Id:#{person.id}, Age:#{person.age}"
+    end
+  end
+
+  def create_person
+    puts 'Do you want a student(1) or a teacher(2)? [input the number]:'
+    choice = gets.chomp.to_i
+    case choice
+    when 1
+      create_student
+    when 2
+      create_teacher
+    else
+      puts 'Incorrect input'
+    end
+  end
+
+  def create_student
+    print 'Age:'
+    age = Integer(gets.chomp)
+    print 'Name:'
+    name = String(gets.chomp)
+    print 'Has parent permission? [Y/N]:'
+    permission = gets.chomp.downcase
+
+    case permission
+    when 'y'
+      student = Student.new(nil, age, name, parent_permission: true)
+    when 'n'
+      student = Student.new(nil, age, name, parent_permission: false)
+    else
+      puts 'Incorrect input'
+      return
+    end
+
+    @people << student
+    puts "\nPerson created successfully"
+  end
+
+  def create_teacher
+    print 'Age :'
+    age = Integer(gets.chomp)
+    print 'Name :'
+    name = String(gets.chomp)
+    print 'Specialization :'
+    specialization = String(gets.chomp)
+    teacher = Teacher.new(specialization, age, name, parent_permission: true)
+    @people << teacher
+    puts 'Person created successfully'
+  end
+
+  def create_book
+    print 'Title :'
+    title = String(gets.chomp)
+    print 'Author :'
+    author = String(gets.chomp)
+    book = Book.new(title, author)
+    @books.push(book)
+    puts "\nBook created successfully"
+  end
+
+  def create_rental
+    puts "\nSelect a book from the following list by number\n"
+    @books.each_with_index do |book, index|
+      puts "#{index}) Title: '#{book.title}', Author: '#{book.author}'"
+    end
+    selected_book = @books[gets.chomp.to_i]
+    puts "\nSelect a person from the following list by number\n"
+    @people.each_with_index do |person, index|
+      puts "#{index}) [#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+    end
+    selected_person = @people[gets.chomp.to_i]
+    puts 'Date (YYYY-MM-DD):'
+    input_date = gets.chomp
+    created_rental = Rental.new(input_date, selected_person, selected_book)
+    @rentals << created_rental
```

Comment:
- [OPTONAL ] Well, I can create the persons and their IDs but one issue that you need to fix is that when I am creating the rental, I can't add the date successfully and that is bothering me from creating the rental.   Though the previous reviewer did not mention it but I recommend you to fix it, so that your project could work as expected.

Reply:
@ishimwezachee  Thank you, bro.I will fix it